src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -35,6 +35,9 @@ namespace Microsoft.CodeAnalysis.CodeActions
     /// </summary>
     public abstract class CodeAction
     {
+        private static readonly ConditionalWeakTable<Type, StrongBox<bool>> s_isNonProgressGetChangedSolutionAsyncOverridden = new();
+        private static readonly ConditionalWeakTable<Type, StrongBox<bool>> s_isNonProgressComputeOperationsAsyncOverridden = new();
```

Comment:
this is an important file to review, and this part is very important.  What we are trying to solve is that we have existing virtual APIs like:

```
public virtual Task<Solution> GetChangedSolution(CancellationToken)
{
    // default impl, that has no progress to pass along.
}
```

// We then want to add this sibling helper:

```
public virtual Task<Solution> GetChangedSolution(IProgress, CancellationToken)
{
    // default impl that passes along progress.
}
```

Which the code action infrastructure will call into.  In the case where the original GetChangedSolution is not overridden, this is good, and we get the right behavior.  However, if GetChangedSolution was overridden, we must respect that override and call into that instead.  So, effectively we do:

```
public virtual Task<Solution> GetChangedSolution(IProgress, CancellationToken)
{
    if (original-get-changed-solution-was-overridden)
        return GetChangedSolution(cancellationToken);

    // otherwise
    // default impl that passes along progress.
}
```

We use this pattern in a few places in roslyn where we need sibling virtuals with differing behavior, but where the new virtual should respect if the original virtual was overridden.